_pages/docs/testing-with-sjs.md

```diff
@@ -60,7 +60,31 @@ Note that the last line returns the `assertions` array.
 
 ### Testing For Expected Errors
 
-*TODO*
+```javascript
+let leftArray = [1,2,3]
+let rightArray = [1,2,3,4]
+
+// Scenario 1: Expected message hard coded
+
+try{
+  test.assertEqual(leftArray, rightArray)
+} catch(err) {
+  test.assertEqual(err.message, "Assert Equal failed:")
+}
+// Scenario 2: Expected message passed in
+let leftArray = [1,2,3]
+let rightArray = [1,2,3,4]
+let message = "Assert Equal failed:"
+
+// Scenario 1: Expected message hard coded
+
+try{
+  test.assertEqual(leftArray, rightArray, message)
+} catch(err) {
+  test.assertEqual(err.message, message)
+}
+Both the scenarios checks left and right arrays and since they are not equal throws an exception. The exception message is validated with assertEqual as part of the test case
+```
 
 ## Testing XQuery using SJS
```

Comment:
I'm not sure we want to encourage testing XQuery using SJS.  If the code is XQuery, I think we might as well encourage writing XQuery tests.  Maybe we can have a simple example just to show that it can be done.

Reply:
If the XQuery code is generating JSON, testing it with SJS is a pretty reasonable thing to do (opinion).